packages/twenty-server/src/workspace/messaging/commands/crons/fetch-all-workspaces-messages.job.ts

```
@@ -38,35 +41,33 @@ export class FetchAllWorkspacesMessagesJob
       })
     ).map((workspace) => workspace.id);
 
-    for (const workspaceId of workspaceIds) {
+    const dataSources = await this.dataSourceRepository.find({
+      where: {
+        workspaceId: In(workspaceIds),
+      },
+    });
+
+    const workspaceIdsWithDataSources = new Set(
+      dataSources.map((dataSource) => dataSource.workspaceId),
+    );
+
+    for (const workspaceId of workspaceIdsWithDataSources) {
       await this.fetchWorkspaceMessages(workspaceId);
     }
   }
 
   private async fetchWorkspaceMessages(workspaceId: string): Promise<void> {
-    try {
```

We don't want to swallow the exception anymore as it should not happen